packages/@ourworldindata/grapher/src/controls/Controls.tsx

```diff
@@ -1,18 +1,500 @@
 import React from "react"
-import { computed, action } from "mobx"
+import { createPortal } from "react-dom"
+import { computed, action, observable } from "mobx"
 import { observer } from "mobx-react"
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome/index.js"
-import { faChevronDown } from "@fortawesome/free-solid-svg-icons"
 import {
+    faXmark,
+    faGear,
+    faInfoCircle,
+    faPencilAlt,
+    faEye,
+    faRightLeft,
+} from "@fortawesome/free-solid-svg-icons"
+import { EntityName } from "@ourworldindata/core-table"
+import { SelectionArray } from "../selection/SelectionArray"
+import { ChartDimension } from "../chart/ChartDimension"
+import {
+    GRAPHER_SETTINGS_DRAWER_ID,
+    ChartTypeName,
     FacetAxisDomain,
     FacetStrategy,
     StackMode,
+    ScaleType,
 } from "../core/GrapherConstants"
 import { AxisConfig } from "../axis/AxisConfig"
-import { Tippy } from "@ourworldindata/utils"
-import { MoreButtonContext } from "./CollapsibleList/CollapsibleList"
+import { Tippy, range } from "@ourworldindata/utils"
 import classnames from "classnames"
 
+const {
+    LineChart,
+    ScatterPlot,
+    StackedArea,
+    StackedDiscreteBar,
+    StackedBar,
+    Marimekko,
+} = ChartTypeName
+
+export type ControlsManager = EntitySelectionManager & SettingsMenuManager
+
+export interface EntitySelectionManager {
+    showSelectEntitiesButton?: boolean
+    showChangeEntityButton?: boolean
+    showAddEntityButton?: boolean
+    entityType?: string
+    entityTypePlural?: string
+    isSelectingData?: boolean
+}
+
+@observer
+export class EntitySelectorToggle extends React.Component<{
+    manager: EntitySelectionManager
+}> {
+    render(): JSX.Element | null {
+        const {
+            showSelectEntitiesButton,
+            showChangeEntityButton,
+            showAddEntityButton,
+            entityType,
+            entityTypePlural,
+            isSelectingData: active,
+        } = this.props.manager
+
+        const [icon, label] = showSelectEntitiesButton
+            ? [faEye, `Select ${entityTypePlural}`]
```

Comment:
Unfortunately, many charts don't specify the proper `entityType` and `entityTypePlural`. `entityTypePlural` in particular is often not set since it's only used for some scatter plots, and `entityType` is mainly used for faceting, so I wouldn't expect this to be set for charts that can't be faceted. We've discussed this [in this thread](https://owid.slack.com/archives/C05KKHZL8LC/p1692105774588569) in a different context. Back then I did a quick analysis and since we don't have many non-country charts, it's only around 120 charts that would need to be updated (maybe more, I don't remember what I checked specifically). Some of these charts could be updated in bulk (e.g. if "egg" is one of the entities, use "product" and "products", etc.) but others would need to be updated manually. I wanted to do that in this cycle but I didn't find the time. It's definitely something we should consider doing when overhauling the entity picker (currently, the heading is "Choose data to show"), but if it's useful for the controls as well, then we might as well do it now. If you do end up doing the data work, it would be nice to also update the heading in the entity picker modal.

Reply:
Agreed that it's probably worth fixing at the data level since we'll be able to make use of the names again in the entity-selector UI.

Just to clarify though: when the config doesn't override the entity type names, they default to `country or region` and `countries or regions`, so there will always be **a** string for this, it will just be the **wrong** string until we update those un-customized, non-country charts.

Reply:
Yes, it would show the default (which for non-country entities is wrong).